MAINTAINERSHIP.md

```diff
@@ -0,0 +1,70 @@
+# Request For Maintainership
+
+This document details the acceptance process for requests from external contributors who wish for the `ibc-go` core team to take over **maintainership** of a complex block of code (ie: an application module or a light client implementation). It is the process we also follow internally for features that we develop which will go into the `ibc-go` codebase.
+
+For projects who have created a custom IBC application and want us to list this application on the registry, please break out your module into its own repo for ease of import into `ibc-go`!
+
+- Create a repo with the custom module in a folder `x/` or `modules/`. 
+- Please include an app that contains the custom module along with end-to-end tests that spins up the blockchain and runs the custom module. 
+- See [cosmos/interchain-security](https://github.com/cosmos/interchain-security) for an example of this setup.
+
+For contributors wishing to submit contributions to the codebase, please check our [Contributor Guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md) :)
+
+
+<p align="center">
+  <img src="maintainership.png?raw=true" alt="maintainership" width="80%" />
+</p>
+
+## Step 1: Product check
+
+Reach out to the IBC product team through susannah@interchain.io to coordinate use-case walkthrough.
+
+Answer these questions in a requirements doc:
+
+```
+    What problem does this feature solve?
+
+    What are the current solutions or workarounds? 
+
+    Are there other versions or implementations?
+
+    If there are other versions of this feature, why is this solution better?
+
+    What are the use cases?
+
+    Which users have confirmed they will use this? 
+
+    How urgent is it to implement this feature?
+
+    How soon after being developed would this be adopted? 
+
+    What is the impact of this feature being adopted?
+```
+
+Answers to these questions should also be detailed in an issue in the `ibc-go` repo to open up the discussion to a wider audience, this can be done before or after the walk through. 
+
+The acceptance criteria is based on the answers to these questions - the feature must solve a genuine problem, have users that would greatly benefit from the solution and be generic enough to benefit many users of the `ibc-go` implementation. 
+
+## Step 2: Submit spec to the IBC protocol repo
+
+A detailed review of the specification can be expected **within 2 weeks** of submission. Please notify the specification team if this does not occur, so it can be corrected as soon as possible.
```

Comment:
Is it worth limiting expectations somewhat here? At the moment this seems like an expectation for @AdityaSripal to review the spec within 2 weeks. What if he is on vacation? Maybe it is better to say the submitter will be given an expected timeline for specification review. I agree it is important to commit to some expected turnaround time

The product check should happen before the spec review and this states "within 2 weeks of submission" which may not give much time for spec review

Reply:
within 2 weeks of submission means 2 weeks of submission for the spec, not the product check, i can clarify that. re: timeline... i can add that this will be communicated explicitly but expected timeline is 2 weeks. how does that sound?